### PR TITLE
Web console: Interval input component

### PR DESCRIPTION
**Pull request by renevan10**
This PR creates a reusable interval input component, where the user can enter the interval manually (like before) or select two dates from a calendar. Each date is enforced to be a standard ISO interval

image

### CODE REVIEW

**Comment by fjy** [comment]
Can we also add in the ability to type HH:mm:ss?

**Comment by renevan10** [comment]
@fjy Sure, what would the format be ? YYYY-MM-DDTHH:mm:ss ?

**Comment by vogievetsky** [comment]
@renevan10 yes, standard ISO interval (I think you already support that right?)

**Comment by fjy** [comment]
Can we clearly label which boxes are for HH, mm, and ss?

**Comment by renevan10** [comment]
@fjy I'm not sure if that's possible, it would require a complex override on the calendar component. I think users should know which ones are which given that the component itself doesn't have those labels?

**Comment by vogievetsky** [comment]
Could you restrict the input to only accept chars that are valid for intervals to have so so not allow this:

image

File: web-console/src/components/interval-input/interval-input.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+export const IntervalInput = React.memo(function IntervalInput(props: IntervalInputProps) {
+  const [tempInterval, setTempInterval] = useState();
+  const { interval, placeholder, onValueChange } = props;
+
+  function removeLocalTimezone(localDate: Date): Date {
+    // Function removes the local timezone of the date and displays it in UTC
+    return new Date(localDate.getTime() - localDate.getTimezoneOffset() * 60000);
+  }
+
+  function parseInterval(interval: string): DateRange {
```
Comment (vogievetsky) [inline web-console/src/components/interval-input/interval-input.tsx]: you can lift this function out of the render context (put it next to CURRENT_YEAR). I think the same can be said for the other functions

File: web-console/src/components/interval-input/interval-input.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        ? removeLocalTimezone(localEndDate)
+            .toISOString()
+            .substring(0, 19)
+        : ''
+    }`;
+  }
+
+  return (
+    <InputGroup
+      value={`${interval || tempInterval}`}
```
Comment (vogievetsky) [inline web-console/src/components/interval-input/interval-input.tsx]: Why the extra string interpolation?

File: web-console/src/components/interval-input/interval-input.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            }
+            position={Position.BOTTOM_RIGHT}
+          >
+            <Button rightIcon={IconNames.CALENDAR} />
+          </Popover>
+        </div>
+      }
+      onChange={(e: any) => {
+        if (
+          (e.target.value.match(/^[\-0-9T:/]+$/g) && e.target.value.length <= 39) ||
```
Comment (vogievetsky) [inline web-console/src/components/interval-input/interval-input.tsx]: This is not what you want, you want to filter here.

File: web-console/src/views/query-view/query-output/query-output.scss [hunk 4]
```diff
@@ -22,6 +22,7 @@
     top: 0;
     bottom: 0;
     width: 100%;
+    font-variant-numeric: tabular-nums;
```
Comment (vogievetsky) [inline web-console/src/views/query-view/query-output/query-output.scss]: look at https://css-tricks.com/almanac/properties/f/font-variant-numeric/

I think you also want to add `font-feature-settings: tnum`

**Comment by vogievetsky** [comment]
Looks great, thank you for addressing the feedback 👍

**Pull request merged**